schema/updates/2022_11_21_SSX_experiment.sql

```diff
@@ -0,0 +1,125 @@
+INSERT INTO SchemaStatus (scriptName, schemaStatus) VALUES ('2022_11_21_SSX_experiment.sql', 'ONGOING');
+
+CREATE TABLE `SSXDataCollection`(
+    `dataCollectionId` int(11) unsigned PRIMARY KEY,
+    `repetitionRate` float DEFAULT NULL,
+    `energyBandwidth` float DEFAULT NULL,
+    `monoStripe` varchar(255) DEFAULT NULL,
+    `jetSpeed` float DEFAULT NULL,
+    `jetSize` float DEFAULT NULL,
+    `chipPattern` varchar(255) DEFAULT NULL,
+    `chipModel` varchar(255) DEFAULT NULL,
+    `reactionDuration` float DEFAULT NULL,
+    `laserEnergy` float DEFAULT NULL,
+    `experimentName` varchar(255) DEFAULT NULL,
+    FOREIGN KEY (`dataCollectionId`) REFERENCES `DataCollection`(`dataCollectionId`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE = InnoDB AUTO_INCREMENT = 0 DEFAULT CHARSET = utf8mb4;
+
+ALTER TABLE
+    `DataCollectionGroup`
+MODIFY
+    COLUMN `experimentType` enum(
+        'EM',
+        'SAD',
+        'SAD - Inverse Beam',
+        'OSC',
+        'Collect - Multiwedge',
+        'MAD',
+        'Helical',
+        'Multi-positional',
+        'Mesh',
+        'Burn',
+        'MAD - Inverse Beam',
+        'Characterization',
+        'Dehydration',
+        'Still',
+        'SSX-Chip',
+        'SSX-Jet'
+    );
+
+ALTER TABLE
+    `ComponentType`
+ADD
+    UNIQUE (`name`);
+
+INSERT INTO
+    `ComponentType` (`name`)
+values
+    ('Ligand'),
+    ('Buffer'),
+    ('JetMaterial');
+
+CREATE TABLE `Component` (
+    `componentId` int(11) unsigned PRIMARY KEY AUTO_INCREMENT,
+    `componentTypeId` int(11) unsigned NOT NULL,
+    `proposalId` int(10) unsigned DEFAULT NULL,
+    `name` varchar (255) NOT NULL,
+    `composition` varchar(255) DEFAULT NULL,
+    `concentration` float DEFAULT NULL,
+    FOREIGN KEY (`componentTypeId`) REFERENCES `ComponentType`(`componentTypeId`),
+    FOREIGN KEY (`proposalId`) REFERENCES `Proposal`(`proposalId`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE = InnoDB AUTO_INCREMENT = 0 DEFAULT CHARSET = utf8mb4;
+
+CREATE TABLE `SampleComposition` (
+    `sampleCompositionId` int(11) unsigned PRIMARY KEY AUTO_INCREMENT,
+    `componentId` int(11) unsigned NOT NULL,
+    `blSampleId` int(11) unsigned NOT NULL,
+    `concentrationTypeId` int(10) unsigned DEFAULT NULL,
+    `abundance` float DEFAULT NULL,
+    `ratio` float DEFAULT NULL,
+    `ph` float DEFAULT NULL,
+    FOREIGN KEY (`componentId`) REFERENCES `Component`(`componentId`),
+    FOREIGN KEY (`blSampleId`) REFERENCES `BLSample`(`blSampleId`),
+    FOREIGN KEY (`concentrationTypeId`) REFERENCES `ConcentrationType`(`concentrationTypeId`)
+) ENGINE = InnoDB AUTO_INCREMENT = 0 DEFAULT CHARSET = utf8mb4;
+
+CREATE TABLE `CrystalComposition` (
+    `crystalCompositionId` int(11) unsigned PRIMARY KEY AUTO_INCREMENT,
+    `componentId` int(11) unsigned NOT NULL,
+    `crystalId` int(11) unsigned NOT NULL,
+    `concentrationTypeId` int(10) unsigned DEFAULT NULL,
```

Comment:
You could also move `concentrationTypeId` up into `Component`. The unit of concentration should be consistent between instances. I.e. NaCl should be measured in Molar, regardless of what the actual value of concentration is (0.1M)

Reply:
I think it makes more sense to store the unit at the same level as the value. It makes things simpler to understand for devs, and also makes more sense from user perspective as it means they will define concentration value and unit at the same time.